source/Halibut.TestUtils.CompatBinary.Base/Utils/DataStreamExtensionMethods.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.IO;
+
+namespace Halibut.TestUtils.SampleProgram.Base.Utils
+{
+    public static class DataStreamExtensionMethods
+    {
+        public static void ConfigureWriterOnReceivedDataStream(this DataStream dataStream)
```

Comment:
Nit: It would be nice if we could return a new DataStream which does exactly what you do below.

This would leave the given data stream alone, which I think is a cleaner implementation.

Reply:
Good idea it does now.